packages/nimble-components/src/select/template.ts

```diff
@@ -22,9 +22,14 @@ import {
     filterSearchLabel
 } from '../label-provider/core/label-tokens';
 import { FilterMode } from './types';
+import { ListOptionGroup } from '../list-option-group';
 import { buttonTag } from '../button';
 import { iconTimesTag } from '../icons/times';
 
+const isListOptionGroup = (n: HTMLElement): boolean => {
```

Comment:
This function is duplicated in a few places in this PR. Can we single-source it?

Also, the[ FAST equivalent `isListboxOption`](https://github.com/microsoft/fast/blob/00bccd7812ca072fc2efdf39af7762bbfdc90846/packages/web-components/fast-foundation/src/listbox-option/listbox-option.ts#L19) doesn't only do an `instanceof` check; it also checks the `role`. This seems more flexible since it would allow slotting new types of elements as long as they meet the API contract. Should we do the same? I think we do that for other element types. Or is our API contract too specific to our group element, so we'll never be able to support something like a native `<optgroup>`?

Reply:
We are being more strict because we use this check to access APIs on the `ListOptionGroup` class (namely the `visuallyHidden` attribute). At the moment, I'm not sure of a better approach.

Reply:
Talked off-line and we're good with using explicit `instanceof` checks for now. I've single-sourced the implementation.